server/athenian/api/controllers/security_controller.py

```diff
@@ -1,14 +1,135 @@
+from athenian.api.models.state.models import UserToken
+from athenian.api.models.web import BadRequestError, ForbiddenError, NotFoundError
+from athenian.api.models.web.create_token_request import CreateTokenRequest
+from athenian.api.models.web.created_token import CreatedToken
+from athenian.api.models.web.generic_error import DatabaseConflict, ServerNotImplementedError
+from athenian.api.models.web.listed_token import ListedToken
+from athenian.api.models.web.patch_token_request import PatchTokenRequest
+from athenian.api.request import AthenianWebRequest
+from athenian.api.response import model_response, ResponseError
+
 
 def info_from_bearerAuth(token: str) -> Optional[Dict[str, Any]]:
     """
-    Check and retrieve authentication information from custom bearer token.
+    Check and retrieve authentication information from a Bearer token.
+
+    Returned value will be passed in 'token_info' parameter of your operation function, if there
+    is one. 'sub' or 'uid' will be set in 'user' parameter of your operation function, if there
+    is one. Should return None if auth is invalid or does not allow access to called API.
+
+    The real work happens in Auth0._set_user().
+    """
+    return {"token": token, "method": "bearer"}
+
+
+def info_from_apiKeyAuth(token: str, required_scopes) -> Optional[Dict[str, Any]]:
+    """
+    Check and retrieve authentication information from an API key.
 
     Returned value will be passed in 'token_info' parameter of your operation function, if there
     is one. 'sub' or 'uid' will be set in 'user' parameter of your operation function, if there
     is one. Should return None if auth is invalid or does not allow access to called API.
 
     The real work happens in Auth0._set_user().
     """
-    return {"token": token}
+    return {"token": token, "method": "apikey"}
+
+
+async def create_token(request: AthenianWebRequest, body: dict) -> web.Response:
+    """Create a new Personal Access Token for the current user and the specified account."""
+    kms = request.app["kms"]
+    if kms is None:
+        raise ResponseError(ServerNotImplementedError(detail="Google KMS was not initialized."))
+    model = CreateTokenRequest.from_dict(body)  # type: CreateTokenRequest
+    _check_token_name(model.name)
+    async with request.sdb.connection() as conn:
+        await get_user_account_status(request.uid, model.account, conn, request.cache)
+        try:
+            token = await conn.execute(insert(UserToken).values(UserToken(
+                name=model.name, account_id=model.account, user_id=request.uid,
+            ).create_defaults().explode()))
+        except (UniqueViolationError, IntegrityError, OperationalError) as e:
+            raise ResponseError(DatabaseConflict(
+                detail="Token '%s' already exists: %s: %s" % (model.name, type(e).__name__, e)),
+            ) from None
+    plaintext = struct.pack("<q", token)
```

Comment:
`"<Q"` should be better since `ids` are positive.

Reply:
Gotcha! `BigInteger` and `Integer` are actually signed and you can have negative IDs - for example, if you create DB records by hand. Anyways, when we reach `1<<63` API keys we'll have other problems :D

Reply:
> for example, if you create DB records by hand.

yeah I just excluded this case as a thing to avoid 😛 , but good point and also since the schema as well is actually integer that's fine 👍 .

> Anyways, when we reach 1<<63 API keys we'll have other problems :D

😂 💯